jwst/cube_build/ifu_cube.py

```diff
@@ -685,28 +681,53 @@ def setup_IFUCube(self,j):
         err_cube = np.zeros((naxis3, naxis2, naxis1))
 
         IFUCube = datamodels.IFUCubeModel(data=data, dq=dq_cube, err=err_cube, weightmap=idata)
-
         IFUCube.update(self.input_models[j])
-
         IFUCube.meta.filename = self.output_name
-
+        
+        self.blend_output_metadata(IFUCube)
+#______________________________________________________________________
         if self.output_type == 'single':
             with datamodels.open(self.input_models[j]) as input:
-
                 # define the cubename for each single
                 filename = self.input_filenames[j]
                 indx = filename.rfind('.fits')
                 self.output_name_base = filename[:indx]
                 self.output_file = None
-                newname  = IFUCubeData.define_cubename(self)
+                newname  = self.define_cubename()
                 IFUCube.meta.filename = newname
-                # need to set what type of SINGLE IFU Cube is being created in header
-                # this information is needed in the blotting stage
-                if(self.instrument == 'MIRI'):
-                    IFUCube.meta.instrument.channel = self.list_par1[0]
-                if(self.instrument == 'NIRSPEC'):
-                    IFUCube.meta.instrument.grating = self.list_par1[0]
 
+#______________________________________________________________________
+# fill in Channel, Band for MIRI
+        if self.instrument == 'MIRI':
+            # fill in Channel output meta data
+            num_ch = len(self.list_par1)
+            IFUCube.meta.instrument.channel = self.list_par1[0]
+            num_ch = len(self.list_par1)
+            for j in range (1, num_ch):
+                IFUCube.meta.instrument.channel =  IFUCube.meta.instrument.channel + str(self.list_par1[j])
+
+            # fill in Band output meta  data
+            num_band = len(set(self.list_par2))
+            if num_band == 1:  
+                IFUCube.meta.instrument.band = self.list_par2[0]
+            else:
+                IFUCube.meta.instrument.band = 'MULTIPLE'
+#______________________________________________________________________
+# fill in Grating, Filter for  NIRSPEC
```

Comment:
Same comment here as for BAND: can remove this code for GRATING and FILTER and just add "blend_rule: multi" to these keywords in core.schema.yaml.